src/Tools/AnalyzerRunner/Program.cs

```diff
@@ -39,169 +37,178 @@ public static async Task Main(string[] args)
-                            var description = existing.description;
-                            if (string.IsNullOrEmpty(description))
-                            {
-                                description = diagnostic.Descriptor?.Title.ToString();
-                                if (string.IsNullOrEmpty(description))
-                                {
-                                    description = diagnostic.Descriptor?.MessageFormat.ToString();
-                                }
-                            }
-
-                            diagnosticStatistics[diagnostic.Id] = (description, diagnostic.Descriptor.DefaultSeverity, existing.count + 1);
+                            description = diagnostic.Descriptor?.MessageFormat.ToString();
                         }
                     }
 
-                    foreach (var pair in diagnosticStatistics)
-                    {
-                        Console.WriteLine($"  {pair.Value.severity} {pair.Key}: {pair.Value.count} instances ({pair.Value.description})");
-                    }
+                    diagnosticStatistics[diagnostic.Id] = (description, diagnostic.Descriptor.DefaultSeverity, existing.count + 1);
                 }
+            }
 
-                Console.WriteLine("Pausing 5 seconds before starting analysis...");
-                await Task.Delay(TimeSpan.FromSeconds(5)).ConfigureAwait(false);
-
-                await incrementalAnalyzerRunner.RunAsync(workspace, cancellationToken).ConfigureAwait(false);
-                await diagnosticAnalyzerRunner.RunAsync(workspace, cancellationToken).ConfigureAwait(false);
-                await codeRefactoringRunner.RunAsync(workspace, cancellationToken).ConfigureAwait(false);
+            foreach (var pair in diagnosticStatistics)
+            {
+                Console.WriteLine($"  {pair.Value.severity} {pair.Key}: {pair.Value.count} instances ({pair.Value.description})");
             }
         }
 
+        private static void ShowSolutionStatistics(Solution solution, CancellationToken cancellationToken)
+        {
+            var sums = new ConcurrentBag<Statistic>();
+            var projects = solution.Projects.Where(project => project.Language == LanguageNames.CSharp || project.Language == LanguageNames.VisualBasic).ToList();
+
+            Console.WriteLine("Number of projects:\t\t" + projects.Count);
+            Console.WriteLine("Number of documents:\t\t" + projects.Sum(x => x.DocumentIds.Count));
+
+            Parallel.ForEach(projects.SelectMany(project => project.Documents), document =>
+            {
+                var documentStatistics = GetSolutionStatisticsAsync(document, cancellationToken).ConfigureAwait(false).GetAwaiter().GetResult();
+                sums.Add(documentStatistics);
+            });
+
+            var statistics = sums.Aggregate(new Statistic(0, 0, 0), (currentResult, value) => currentResult + value);
```

Comment:
Why is this not just calling GetSolutionStatistics?

```suggestion
            var statistics = GetSolutionStatistics(projects, cancellationToken);
```

Reply:
Fixed